frontend/src/stores/form.js

```diff
@@ -0,0 +1,34 @@
+import { defineStore } from 'pinia';
+import { objectDeepCopy } from '@/helpers/object';
+import { useStorage } from '@vueuse/core';
+
+const defaultSubjectFormState = {
+  name: '',
+  timesPerWeek: 0,
+  teacher: { name: '' },
+  classroom: { name: '' },
+};
+
+const defaultClassFormState = {
+  name: '',
+  subjects: [defaultSubjectFormState],
+};
+
+const defaultFormState = {
+  classes: [defaultClassFormState],
+};
```

Comment:
if these were factories that returned an object there would not be any need for deep cloning, especially since they are never used outside of deep cloning.

Reply:
I haven't tried that but I think it also won't work, nested objects in JS are weird. See my response to last comment.

Reply:
Please try something like this because I believe it will work since it will create new objects every time it is called:
```js
const buildSubjectFormState = () => ({
  name: '',
  timesPerWeek: 0,
  teacher: { name: '' },
  classroom: { name: '' },
});

const buildClassFormState = () => ({
  name: '',
  subjects: [buildSubjectFormState()],
});

const buildFormState = () => {
  classes: [buildClassFormState()],
};
```

and then used like this:
```js
export const useFormStore = defineStore('form', () => {
  const form = useStorage('form', buildFormState());
  const accordionState = useStorage('accordionState', {});

  function addClass() {
    form.value.classes.push(buildClassFormState());
  }

  function addSubject(index) {
    form.value.classes[index].subjects.push(buildSubjectFormState());
  }

  return { form, accordionState, addClass, addSubject };
});
```